[cm] Implement identifier pool module

mlohvynenko opened the pull request:
This patch implements cm identifier pool module,
that is used to manage unique identifiers
for various resources in the system, such as UIDs and GIDs.

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

mlohvynenko committed:
gitignore: ignore cmake user presets file

This patch updates the .gitignore file to ignore
the CMakeUserPresets.json file.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
common: tools: add identifier pool

This patch adds a new identifier pool utility to the common
tools library.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko commented on include/aos/cm/identifierpool/identifierpool.hpp:
```diff
@@ -0,0 +1,103 @@
+     *
+     * @return RetWithError<size_t>.
+     */
+    virtual RetWithError<size_t> GetFreeID() = 0;
```
applied suggestion

mlohvynenko commented on include/aos/cm/identifierpool/identifierpool.hpp:
```diff
@@ -0,0 +1,103 @@
+/**
+ * Interface for identifier pool.
+ */
+class IdentifierPoolItf : public NonCopyable {
```
applied suggestion

al1img reviewed (approved):
`Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>`

MykolaSuperman reviewed (approved):
`Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>`
